plugins/woocommerce-admin/docs/features/onboarding-tasks.md

```diff
@@ -85,29 +134,138 @@ $args = array(
+- `$task_list::is_hidden()`: Checks if the task list is hidden.
+- `$task_list::is_complete()`: Checks if all tasks in the task list are complete.
+- `$task_list::get_completed_count()`: Returns the count of completed tasks in the task list.
+- `$task_list::get_total_count()`: Returns the total count of tasks in the task list.
+- `$task_list::get_progress_percentage()`: Returns the progress percentage of the task list.
+- `$task_list::get_sorted_tasks($sort_by)`: Returns the tasks sorted based on the specified sorting criteria.
+- `$task_list::get_json()`: Returns the JSON representation of the task list.
+- `$task_list->get_json()` - Get the camelcase JSON for use in the client
+  - `id` (int) - Task list ID.
+  - `title` (string) - Task list title.
+  - `isHidden` (bool) - If a task has been hidden.
+  - `isVisible` (bool) - If a task list is visible.
+  - `isComplete` (bool) - Whether or not all viewable tasks have been completed.
+  - `tasks` (array) - An array of `Task` objects.
+
+### Task
+
+The `Task` class represents a task. It contains properties and methods for managing tasks. You can see the predefined tasks in [this directory](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks).
+
+Please note that the `Task` class is abstract and intended to be extended by custom task classes.
+
+#### Example
+
+```php
+<?php
+
+use Automattic\WooCommerce\Admin\Features\OnboardingTasks\Task;
+
+class MyTask extends Task {
+    public function get_id() {
+        // Return a unique identifier for your task
+    }
+
+    public function get_title() {
+        // Return the title of your task
+    }
+
+    public function get_content() {
+        // Return the content/explanation of your task
+    }
+
+    public function get_time() {
+        // Return the estimated time to complete the task
+    }
+    // Implement other abstract methods as needed
+}
+
+// Add MyTask to "test-list" task list
+TaskLists::add_task(
+  'test-list',
+  new MyTask(
+    TaskLists::get_list( 'test-list' )
+  )
+);
+```
+### Methods
+
+- `$task->get_id(): string`: Returns the ID of the task.
+- `$task->get_title(): string`: Returns the title of the task.
+- `$task->get_content(): string`: Returns the content of the task.
+- `$task->get_time(): string`: Returns the estimated time to complete the task.
+- `$task->get_parent_id(): string`: Returns the ID of the parent task list.
+- `$task->get_parent_options(): array`: Returns the options of the parent task list.
+- `$task->get_parent_option($option_name): mixed|null`: Returns the value of a specific option from the parent task list.
+- `$task->prefix_event($event_name): string`: Returns the event name prefixed with the task list's event prefix.
+- `$task->get_additional_info(): string`: Returns additional information about the task.
+- `$task->get_additional_data(): mixed|null`: Returns additional data associated with the task.
```

Comment:
I think it'll be good for us to clarify the difference between these two.

```suggestion
- `$task->get_additional_info(): string`: Returns additional information about the task. Typically includes details, notes, or instructions related to the task itself.
- `$task->get_additional_data(): mixed|null`: Returns additional data associated with the task. It can be any type of data, such as arrays, objects, or simple values.
```